aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -0,0 +1,132 @@
+#include <ATen/Context.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/Dispatch.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/cuda/PinnedMemoryAllocator.h>
+#include <ATen/cuda/CUDAApplyUtils.cuh>
+#include <ATen/cuda/detail/IndexUtils.cuh>
+#include <ATen/cuda/CUDASolver.h>
+#include <ATen/cuda/CUDABlas.h>
+#include <ATen/cuda/CUDAEvent.h>
+#include <c10/cuda/CUDAStream.h>
+
+#include <ATen/native/LinearAlgebraUtils.h>
+#include <ATen/native/cuda/MiscUtils.h>
+#include <ATen/native/cuda/BatchLinearAlgebraLib.h>
+
+#ifdef USE_CUSOLVER
+
+namespace at {
+namespace native {
+
+inline static Tensor column_major_identity_matrix_like(const Tensor& self) {
+  int n = cuda_int_cast(self.size(-2), "self.size(-2)");
+  Tensor self_inv = at::eye({n}, self.options()).expand_as(self).contiguous();
+  self_inv.unsafeGetTensorImpl()->set_stride(self.dim()-2, 1); // These two lines set self_inv to column-major
+  self_inv.unsafeGetTensorImpl()->set_stride(self.dim()-1, n);
+  return self_inv;
+}
+
+template <typename scalar_t>
+inline static void _apply_single_inverse_helper(scalar_t* self_ptr, scalar_t* self_inv_ptr, int* ipiv_ptr, int* info_ptr, int n) {
+  // self_inv_ptr should already be an identity matrix
+
+  auto handle = at::cuda::getCurrentCUDASolverDnHandle();
+  at::cuda::solver::getrf<scalar_t>(handle, n, n, self_ptr, n, ipiv_ptr, info_ptr);
+  at::cuda::solver::getrs<scalar_t>(handle, n, n, self_ptr, n, ipiv_ptr, self_inv_ptr, n, info_ptr);
+}
+
+template <typename scalar_t>
+static void apply_batched_inverse_lib(Tensor& self, Tensor& self_inv, Tensor& infos) {
+  const int batch_size = cuda_int_cast(batchCount(self), "batchCount");
+  const int n = cuda_int_cast(self.size(-2), "self.size(-2)");
+
+  auto self_data = self.data_ptr<scalar_t>();
+  auto self_mat_stride = matrixStride(self);
+  auto self_inv_data = self_inv.data_ptr<scalar_t>();
+  auto self_inv_mat_stride = matrixStride(self_inv);
+
+  Tensor self_array = at::empty({batch_size}, at::device(at::kCPU).dtype(at::kLong));
+  Tensor self_inv_array = at::empty({batch_size}, at::device(at::kCPU).dtype(at::kLong));
+  scalar_t** p_self_array = reinterpret_cast<scalar_t**>(self_array.data_ptr());
```

Comment:
for loop launch, you can compute pointers directly in the loop, for batched launch it makes sense to initialize with linspace directly on cuda?

Reply:
loop launch: sure

batched launch: The problem is that the matrix arguments are passed in as "device array of device pointers" ([doc](https://docs.nvidia.com/cuda/cublas/index.html#cublas-lt-t-gt-getrsbatched)). Such array doesn't exist naturally in a tensor, and we have to construct it manually. I guess I can't directly write on a CUDA array with value, so it is first initialized on CPU and then copied to device. 

One better approach is that we can try `arange` with raw pointer address to initialize such array directly on CUDA, and that could save some time.

Reply:
Yeah, that was my suggestion, you could use `arange` or `linspace` functions